alibi_detect/saving/loading.py

```diff
@@ -216,14 +215,20 @@ def _load_preprocess_config(cfg: dict,
             emb = kwargs.pop('embedding')  # embedding passed to preprocess_drift as `model` therefore remove
 
             # Backend specifics
-            if backend == 'tensorflow':
+            if has_tensorflow and isinstance(model, SupportedModels_tf):
```

Comment:
By the time we reach here, `model` has already been resolved. Hence we cannot read the `flavour` field in `ModelConfig` in order to determine whether it is a `tensorflow`, `pytorch` or `sklearn` model. Instead we infer this by examining the model type. The `has_tensorflow` check is to avoid calling `isinstance` when `SupportedModels_tf` is empty.